tests.py

```diff
@@ -281,6 +281,29 @@ def test_no_new_reference_cycles(self):
         self.assertEqual(len(gc.get_referrers(x)), 1)
 
 
+class TypestatsFilterArguTest(GarbageCollectedMixin, unittest.TestCase):
+    """Tests for the typestats function, especially for augument
+    ``fiter`` which is added at version 3.1.3"""
+
+    def test_without_filter(self):
+        MyClass = type('MyClass', (), {'__module__': 'mymodule'})  # noqa
+        x, y = MyClass(), MyClass()
+        x.magic_attr = True
+        y.magic_attr = False
+        stats = objgraph.typestats(shortnames=False)
+        self.assertEqual(2, stats['mymodule.MyClass'])
+
+    def test_with_filter(self):
+        MyClass = type('MyClass', (), {'__module__': 'mymodule'})  # noqa
+        x, y = MyClass(), MyClass()
+        x.magic_attr = True
+        y.magic_attr = False
+        stats = objgraph.typestats(
+            shortnames=False,
+            filter=lambda e: hasattr(e, 'magic_attr') and e.magic_attr)
```

Comment:
This fails on Python 3.4 on Windows because, apparently, `hasattr(ctypes.cdll, 'magic_attr')` tries to load a `magic_attr.dll` and raises an OSError.  Amazing.

I suggest replacing the `hasattr` check with an `isinstance(e, MyClass)` check.

Reply:
Thanks very much. I had run the test on my computer, windows, python3.4 and no error .